src/github.com/mozilla-services/pushgo/simplepush/router_broadcast.go

```diff
@@ -460,27 +440,28 @@ func (r *BroadcastRouter) notifyAll(cancelSignal <-chan bool, contacts []string,
 func (r *BroadcastRouter) notifyBucket(cancelSignal <-chan bool, contacts []string,
 	uaid string, segment *capn.Segment, logID string) (ok bool, err error) {
 
-	result, stop := make(chan bool), make(chan struct{})
-	defer close(stop)
 	timeout := r.ctimeout + r.rwtimeout + 1*time.Second
+	results := make(chan bool, len(contacts))
 	for _, contact := range contacts {
 		url := fmt.Sprintf("%s/route/%s", contact, uaid)
-		go r.notifyContact(result, stop, url, segment, logID)
-	}
-	timer := time.NewTimer(timeout)
-	select {
-	case ok = <-r.closeSignal:
-		return false, io.EOF
-	case <-cancelSignal:
-	case ok = <-result:
-	case <-timer.C:
+		go r.notifyContact(results, url, segment, logID)
+	}
+	timer := time.After(timeout)
+	for i := 0; i < cap(results) && !ok; i++ {
```

Comment:
This actually represents whether it was delivered or not, so it should be 'delivered', not 'ok'.